### PR TITLE
Move macos CI to arm

### PR DESCRIPTION
**Pull request by adrianinsaval**
Github's Mac M1 runners are now available to the public, I think it's more relevant to test there than on intel for mac (plus I think those runners are more powerful). And who knows, maybe those fake failures on mac will go away

@oursland would you have any idea why one of the tests fails here?

### CODE REVIEW

**Comment by oursland** [comment]
Yes, the ARM macOS runners will be substantially faster and better reflect the challenging edge-cases for CI.

It looks like the result of the test `femtest.app.test_solver_calculix.TestSolverCalculix.test_material_multiple_bendingbeam_fiveboxes` resulted in a different answer than expected:

Got `-8.57709328118` and expected `-8.577093281179`.

Full error:

```
======================================================================
FAIL: test_material_multiple_bendingbeam_fiveboxes (femtest.app.test_solver_calculix.TestSolverCalculix.test_material_multiple_bendingbeam_fiveboxes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/runner/work/FreeCAD/FreeCAD/.conda/freecad/opt/freecad/Mod/Fem/femtest/app/test_solver_calculix.py", line 294, in test_material_multiple_bendingbeam_fiveboxes
    self.input_file_writing_test(get_namefromdef("test_"))
  File "/Users/runner/work/FreeCAD/FreeCAD/.conda/freecad/opt/freecad/Mod/Fem/femtest/app/test_solver_calculix.py", line 381, in input_file_writing_test
    self.assertFalse(
AssertionError: 'Comparing /Users/runner/work/FreeCAD/FreeCAD/.conda/freecad/opt/freecad/Mod/Fem/femtest/data/calculix/material_multiple_bendingbeam_fiveboxes.inp to /var/folders/1k/qq3pcbf12vb6vyblh81736p40000gn/T/FEM_unittests/solver_calculix_material_multiple_bendingbeam_fiveboxes_601e3ba11755/Mesh.inp failed!\n--- \n+++ \n@@ -29117 +29117 @@\n-2656,3,-8.577093281179\n+2656,3,-8.57709328118\n' is not false : CalculiX write_inp_file for material_multiple_bendingbeam_fiveboxes test failed.
Comparing /Users/runner/work/FreeCAD/FreeCAD/.conda/freecad/opt/freecad/Mod/Fem/femtest/data/calculix/material_multiple_bendingbeam_fiveboxes.inp to /var/folders/1k/qq3pcbf12vb6vyblh81736p40000gn/T/FEM_unittests/solver_calculix_material_multiple_bendingbeam_fiveboxes_601e3ba11755/Mesh.inp failed!
--- 
+++ 
@@ -29117 +29117 @@
-2656,3,-8.577093281179
+2656,3,-8.57709328118


----------------------------------------------------------------------
Ran 1238 tests in 42.944s

FAILED (failures=1, skipped=2)
```

As for the underlying cause, I am uncertain.  I can look to see if any dependencies changed that could affect CalculiX's solution.

Is it possible that the result is due to differences in multiplications, such as one architecture always employing fused multiply-adds and the other not doing so?  Furthermore, is this error (`0.000000000001` or `1e-12`) valid, or are the constraints too tight?

**Comment by marioalexis84** [comment]
We need to use another comparison strategy. File strings are currently being compared. We need to move to a rounded float value comparison.

**Comment by marioalexis84** [comment]
Related to https://github.com/FreeCAD/FreeCAD/issues/7188

**Comment by adrianinsaval** [comment]
I think the test does a full file comparison. @wwmayer do you know a way to make this test less strict?

**Comment by wwmayer** [comment]
A better idea could be to load the files and then compare the FE meshes.
